common/test/acceptance/pages/lms/courseware.py

```diff
@@ -310,3 +317,247 @@ def get_selected_tab_content(self):
+        """
+
+        # Get the index of the item in the sequence
+        all_items = self.sequence_items
+
+        try:
+            seq_index = all_items.index(vertical_title)
+
+        except ValueError:
+            msg = "Could not find sequential '{0}'.  Available sequentials: [{1}]".format(
+                vertical_title, ", ".join(all_items)
+            )
+            self.warning(msg)
+
+        else:
+
+            # Click on the sequence item at the correct index
+            # Convert the list index (starts at 0) to a CSS index (starts at 1)
+            seq_css = "ol#sequence-list>li:nth-of-type({0})>.nav-item".format(seq_index + 1)
+            self.q(css=seq_css).first.click()
+            # Click triggers an ajax event
+            self.wait_for_ajax()
+
+    # TODO: TNL-6546: Remove method, outline no longer on courseware page
+    def _section_titles(self):
+        """
+        Return a list of all section titles on the page.
+        """
+        chapter_css = '.course-navigation .chapter .group-heading'
+        return self.q(css=chapter_css).map(lambda el: el.text.strip()).results
+
+    # TODO: TNL-6546: Remove method, outline no longer on courseware page
+    def _subsection_titles(self, section_index):
+        """
+        Return a list of all subsection titles on the page
+        for the section at index `section_index` (starts at 1).
+        """
+        # Retrieve the subsection title for the section
+        # Add one to the list index to get the CSS index, which starts at one
+        subsection_css = (
+            ".course-navigation .chapter-content-container:nth-of-type({0}) "
+            ".menu-item a p:nth-of-type(1)"
+        ).format(section_index)
+
+        # If the element is visible, we can get its text directly
+        # Otherwise, we need to get the HTML
+        # It *would* make sense to always get the HTML, but unfortunately
+        # the open tab has some child <span> tags that we don't want.
+        return self.q(
+            css=subsection_css
+        ).map(
+            lambda el: el.text.strip().split('\n')[0] if el.is_displayed() else el.get_attribute('innerHTML').strip()
+        ).results
+
+    # TODO: TNL-6546: Remove method, outline no longer on courseware page
+    def _on_section_promise(self, section_title, subsection_title):
+        """
+        Return a `Promise` that is fulfilled when the user is on
+        the correct section and subsection.
+        """
+        desc = "currently at section '{0}' and subsection '{1}'".format(section_title, subsection_title)
+        return EmptyPromise(
+            lambda: self.is_on_section(section_title, subsection_title), desc
+        )
+
+    def go_to_outline(self):
+        """
+        Navigates using breadcrumb to the course outline on the course home page.
+        """
+        self.q(css='.path a').click()
+
+    @unguarded
+    def is_on_section(self, section_title, subsection_title):
```

Comment:
@jzoldak This may be part of the problem.  

Here are some related questions:

1. Does this method need to be `@unguarded` and is this a good pattern in the old `_on_section_promise` (above) and the new `_wait_for_course_section` method?  Or, is this where the problem lies?
2. The new `def go_to_section`, the one without the TODO to be deleted, is the one that is sometimes ending on the wrong page, never making it to the page reload.